.azure-pipelines.yml

```diff
@@ -55,12 +55,28 @@ jobs:
       - script: |
           pytest
         displayName: "PyTest"
-        condition: eq(variables['RUN_COVERAGE'], 'no')
+        condition: and(eq(variables['RUN_COVERAGE'], 'no'), eq(variables['PRERELEASE_DEPENDENCIES'], 'no'))
 
       - script: |
           pytest --cov --cov-report=xml --cov-context=test
         displayName: "PyTest (coverage)"
-        condition: eq(variables['RUN_COVERAGE'], 'yes')
+        condition: and(eq(variables['RUN_COVERAGE'], 'yes'), eq(variables['PRERELEASE_DEPENDENCIES'], 'no'))
+
+      # TODO: fix all the exceptions here
+      # replacing “:initializing view as actual” and “:Transforming to str index”
+      # with “::anndata._warnings.ImplicitModificationWarning” breaks,
+      # because we’re neither using editable installs nor a src/ layout
+      - script: >
+          pytest
+          -W error
+          -W 'default:initializing view as actual:UserWarning'
+          -W 'default:Transforming to str index:UserWarning'
+          -W 'default:Observation names are not unique. To make them unique:UserWarning'
+          -W 'default:Variable names are not unique. To make them unique:UserWarning'
+          -W 'default::scipy.sparse._base.SparseEfficiencyWarning'
+          -W 'default::dask.array.core.PerformanceWarning'
```

Comment:
If these were in `filterwarnings` and we wouldn’t have to deal with `ImportPathMismatchError`s, these would be

```
filterwarnings = [
    'error',
    'default::anndata._warnings.ImplicitModificationWarning',
    'default:.*names are not unique. To make them unique:UserWarning',
    'default::scipy.sparse._base.SparseEfficiencyWarning',
    'default::dask.array.core.PerformanceWarning',
]
```

All of them have at least like 60 instances (sure, some are parametrized tests) and would therefore need a bit of work to be fixed.

Not in this PR.